parsers/AR.py

```diff
@@ -0,0 +1,563 @@
+            item = float(item.replace(',','.'))
+        formatted.append(item)       
+
+    return formatted
+
+
+def get_thermal():
+    """
+    Requests thermal generation data then parses and sorts by type.  Nuclear is included.
+    Returns a dictionary.
+    """
+
+    #Need to persist session in order to get ControlID and ReportSession so we can send second request
+    #for table data.  Both these variables change on each new request.
+    s = requests.Session()
+    r = s.get(url)
+    pat = re.search("ControlID=[^&]*", r.text).group()
+    spat = re.search("ReportSession=[^&]*", r.text).group()
+    cid = pat.rpartition('=')[2]
+    rs = spat.rpartition('=')[2]
+    full_table = []
+
+    #'En Reserva' plants are not generating and can be ignored.
+
+    for pagenumber in range(1,8,1):
+        t = s.get(turl, params = {'ControlID': cid, 'ReportSession': rs, 'PageNumber': '{}'.format(pagenumber)})
+        text_only = webparser(t)
+        if 'En Reserva' in text_only:
+            break
+        full_table.append(text_only)
+
+    data = list(itertools.chain.from_iterable(full_table))
+    formatted_data = dataformat(data)
+    mapped_data = [power_plant_type.get(x,x) for x in formatted_data]
+    
+    find_totals = [i+1 for i,x in enumerate(mapped_data) if x == 'Totales ']
+    total_thermal_generation = sum([mapped_data[i] for i in find_totals])
+    find_nuclear = [i+2 for i, x in enumerate(mapped_data) if x == 'nuclear']
+    total_nuclear_generation = sum([mapped_data[i] for i in find_nuclear])
+    #find_oil = [i+2 for i, x in enumerate(mapped_data) if x == 'oil']
+    #total_oil_generation = sum([mapped_data[i] for i in find_oil])
+
+    #Assume thermal generation is gas unless specifically mapped to another type.
+    total_gas_generation = total_thermal_generation - total_nuclear_generation
+
+    return {'gas': total_gas_generation, 'nuclear': total_nuclear_generation}
+
+
+
+def get_hydro():
+    """Requests hydro generation data then parses, returns a dictionary."""
+
+    s = requests.Session()
+    r = s.get(hurl)
+    pat = re.search("ControlID=[^&]*", r.text).group()
+    spat = re.search("ReportSession=[^&]*", r.text).group()
+    cid = pat.rpartition('=')[2]
+    rs = spat.rpartition('=')[2]
+    full_table = []
+    
+
+    for pagenumber in range(1,2,1):
+        t = s.get(thurl, params = {'ControlID': cid, 'ReportSession': rs, 'PageNumber': '{}'.format(pagenumber)})
+        text_only = webparser(t)
+        if 'En Reserva' in text_only:
+            break
+        full_table.append(text_only)
+
+    data = list(itertools.chain.from_iterable(full_table))
+    formatted_data = dataformat(data)
+    #mapped_data = [power_plant_type.get(x,x) for x in formatted_data]
+    find_hydro = [i+1 for i,x in enumerate(formatted_data) if x == 'Totales ']
+    total_hydro_generation = sum([formatted_data[i] for i in find_hydro])
+
+    return {'hydro': total_hydro_generation}
```

Comment:
Assuming you mean pumped storage?  There's Rio Grande & Los Reyunos but I haven't found data for them yet.